criando-uma-aplicacao/linha-de-comando.md

```diff
@@ -0,0 +1,734 @@
+}
+
+func extractWinner(userInput string) string {
+    return strings.Replace(userInput, " venceu", "", 1)
+}
+```
+
+O teste agora vai passar.
+
+* `Scanner.Scan()` vai ler até o carácter de nova linha.
+* Só então usamos `Scanner.Text()` para returnar a `string` lida pelo scanner.
+
+Agora que temos alguns testes passando, devemos amarrar isso ao nosso `main`. Lembre-se que devemos sempre almejar ter
+o código funcionando totalmente integrado o mais rápido que pudermos.
+
+No `main.go` adicione o seguinte e execute. \(você pode ter que ajustar o caminho da segunda dependência para refletir 
+o que tem no seu computador\)
+
+```go
+package main
+
+import (
+    "fmt"
+    "github.com/larien/learn-go-with-tests/linha-de-comando/v3"
+    "log"
+    "os"
+)
+
+const dbFileName = "game.db.json"
+
+func main() {
+    fmt.Println("Vamos jogar poker")
+    fmt.Println("Digite {Nome} venceu para registrar uma vitoria")
+
+    db, err := os.OpenFile(dbFileName, os.O_RDWR|os.O_CREATE, 0666)
+
+    if err != nil {
+        log.Fatalf("falha ao abrir %s %v", dbFileName, err)
+    }
+
+    store, err := poker.NewFileSystemPlayerStore(db)
+
+    if err != nil {
+        log.Fatalf("falha ao criar sistema de arquivos para armazenar jogadores, %v ", err)
+    }
+
+    game := poker.CLI{store, os.Stdin}
+    game.PlayPoker()
+}
+```
+
+Você deve receber um erro
+
+```text
+linha-de-comando/v3/cmd/cli/main.go:32:25: implicit assignment of unexported field 'playerStore' in poker.CLI literal
+linha-de-comando/v3/cmd/cli/main.go:32:34: implicit assignment of unexported field 'in' in poker.CLI literal
+```
+
+O que está acontecendo é que por causa da tentativa de associar os campos `playerStore` e `in` na `CLI`. Eles são campos 
+não exportados\(privados\). Nós _podemos_ fazer isso nos nossos testes porque o teste está no mesmo pacote da `CLI` \(`poker`\). 
+Mas nosso `main` é um pacote `main` portanto não tem acesso.
+
+Isso enfatiza a importância de _integrar seu código_. Nós definimos corretamente as dependências da `CLI` como privada 
+\(porque não queremos expô-las para os usuários da `CLI`\) mas não criamos uma forma para os usuário construí-las.  
+
+Existe alguma forma de identificarmos esse problema antes?
+
+### `package mypackage_test`
+
+Nos exemplos usados até agora, quando nós fazemos um arquivo para testes nós declaramos ele como pertencendo ao mesmo pacote
+que estamos testando.
+
+Tudo bem e fazer isso significa no pior dos casos que queremos testar algo que é pertecente somente aquele pacote
+conseguimos acesso aos tipos não exportados.
+
+Mas considerando que, _em geral_, advogamos para _não_ se fazer testes de coisas internas, como Go pode garantir isso?
+ E se pudéssemos testar nosso código aonde somente temos acesso aos tipos exportados \(como em nossp `main`\)?
+
+Quando você escreve um project com múltiplos pacotes eu recomendo fortmente que o nome to seu pacote tenha o sufixo `_test`.
```

Comment:
- `projeto` ao invés de `project` 
- `fortemente` ao invés de `fortmente`
- `nome do seu` ao invés de `nome to seu`